### PR TITLE
Fix spi targetlist contain NULL

### PR DESCRIPTION
**Pull request by zhangh43**

### CODE REVIEW

File: src/pyclient/plpy_spi.c [hunk 1]
```diff
@@ -419,16 +419,20 @@ PLy_spi_execute_plan(PyObject *ob, PyObject *list, long limit) {
         pydict = PyDict_New();
 
         for (j = 0; j < result->res->cols; j++) {
-            pyval = result->args[j].conv.inputfunc(result->res->data[i][j].value,
+        	    if(result->res->data[i][j].isnull) {
+        	        PyDict_SetItemString(pydict, result->res->names[j], Py_None);
+        	    } else {
+                pyval = result->args[j].conv.inputfunc(result->res->data[i][j].value,
                                                    &result->args[j]);
```
Comment (HaozhouWang) [inline src/pyclient/plpy_spi.c]: indent

**Comment by HaozhouWang** [comment]
+1

File: src/pyclient/plpy_spi.c [hunk 2]
```diff
@@ -419,16 +419,20 @@ PLy_spi_execute_plan(PyObject *ob, PyObject *list, long limit) {
         pydict = PyDict_New();
 
         for (j = 0; j < result->res->cols; j++) {
```
Comment (paul-guo-) [inline src/pyclient/plpy_spi.c]: Need to modify for PLy_spi_execute_query() also.

Comment (zhangh43) [reply src/pyclient/plpy_spi.c]: also fixed in PLy_spi_execute_query

Comment (paul-guo-) [reply src/pyclient/plpy_spi.c]: Need to add one test also.

**Pull request merged**